src/main/java/dev/ai4j/openai4j/OpenAiClient.java

```diff
@@ -6,189 +6,223 @@
-            if (connectTimeout == null) {
-                throw new IllegalArgumentException("connectTimeout cannot be null");
-            }
-            this.connectTimeout = connectTimeout;
-            return (B) this;
-        }
-
-        public B readTimeout(Duration readTimeout) {
-            if (readTimeout == null) {
-                throw new IllegalArgumentException("readTimeout cannot be null");
-            }
-            this.readTimeout = readTimeout;
-            return (B) this;
-        }
-
-        public B writeTimeout(Duration writeTimeout) {
-            if (writeTimeout == null) {
-                throw new IllegalArgumentException("writeTimeout cannot be null");
-            }
-            this.writeTimeout = writeTimeout;
-            return (B) this;
-        }
-
-        public B proxy(Proxy.Type type, String ip, int port) {
-            this.proxy = new Proxy(type, new InetSocketAddress(ip, port));
-            return (B) this;
-        }
-
-        public B proxy(Proxy proxy) {
-            this.proxy = proxy;
-            return (B) this;
-        }
-
-        public B logRequests() {
-            return logRequests(true);
-        }
-
-        public B logRequests(Boolean logRequests) {
-            if (logRequests == null) {
-                logRequests = false;
-            }
-            this.logRequests = logRequests;
-            return (B) this;
-        }
-
-        public B logResponses() {
-            return logResponses(true);
-        }
-
-        public B logResponses(Boolean logResponses) {
-            if (logResponses == null) {
-                logResponses = false;
-            }
-            this.logResponses = logResponses;
-            return (B) this;
-        }
-
-        public B logStreamingResponses() {
-            return logStreamingResponses(true);
-        }
-
-        public B logStreamingResponses(Boolean logStreamingResponses) {
-            if (logStreamingResponses == null) {
-                logStreamingResponses = false;
-            }
-            this.logStreamingResponses = logStreamingResponses;
-            return (B) this;
-        }
+  public abstract SyncOrAsyncOrStreaming<CompletionResponse> completion(
+    CompletionRequest request
+  );
+
+  public abstract SyncOrAsyncOrStreaming<String> completion(String prompt);
+
+  public abstract SyncOrAsyncOrStreaming<ChatCompletionResponse> chatCompletion(
+    ChatCompletionRequest request
+  );
+
+  public abstract SyncOrAsyncOrStreaming<String> chatCompletion(
+    String userMessage
+  );
+
+  public abstract SyncOrAsync<EmbeddingResponse> embedding(
+    EmbeddingRequest request
+  );
+
+  public abstract SyncOrAsync<List<Float>> embedding(String input);
+
+  public abstract SyncOrAsync<ModerationResponse> moderation(
+    ModerationRequest request
+  );
+
+  public abstract SyncOrAsync<ModerationResult> moderation(String input);
+
+  public abstract SyncOrAsync<ImageResponse> imagesGenerations(
```

Comment:
it was named after the OpenAI API endpoint `images/generations`, renamed to `imagesGeneration`, since the images count is configurable within request.